### PR TITLE
Add pull request and issue templates to project.

### PR DESCRIPTION
**Pull request by stephsamson**
This PR adds templates to pull requests and issues similar to what we have on the [python-poetry/poetry](https://github.com/python-poetry/core) project.

### CODE REVIEW

**Comment by abn** [comment]
@stephsamson should be support issues for this project? Can we instead focus on managing issues in poetry? Might be easier for the community.

**Comment by abn** [comment]
@stephsamson as per discussions on discord; we can change this one to add a single config that has a link to poetry's issue tracker with "blank issues" disabled.

**Comment by stephsamson** [comment]
@abn Thanks for your feedback; I've worked that in.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Pull Request Check List
+
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion

<!--
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles!
+
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:4]:
```suggestion
--->

```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+- [ ] Added **tests** for changed code.
+- [ ] Updated **documentation** for changed code.
+
+Resolves: <-- add issue number/link here -->
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:8]:
```suggestion
Resolves: python-poetry#<-- add issue number/link here -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion

<!--
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+**Note**: If your Pull Request introduces a new feature or changes the current behavior, it should be based
+on the `develop` branch. If it's a bug fix or only a documentation update, it should be based on the `master` branch.
+
+If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: -->

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (abn) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Probably also move this to before the checklist?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Pull Request Check List
+
+<!-- This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes. But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles! -->
+
+Resolves: python-poetry#<!-- add issue number/link here -->
```
Comment (abn) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Can we move this above the check list title please?

**Pull request merged**